frappe/core/doctype/user/test_user.py

```diff
@@ -235,6 +236,43 @@ def test_rate_limiting_for_reset_password(self):
 
 		self.assertRaises(frappe.ValidationError, user.reset_password, False)
 
+	def test_rate_limiting_for_sign_up(self):
+		delete_password_reset_cache()
+
+		username = 'Test User {0}'
+		prefix = 'test-rate-limiting'
+		email = prefix + '{0}@example.com'
+		limit = 301  # the hardcoded rate limit is 300
+
+		frappe.db.sql("commit")
+		try:
+			frappe.flags.in_import = True  # disable throttling
+			frappe.db.sql("begin")
+			for x in range(limit):
+				frappe.get_doc(dict(
+					doctype='User',
+					email=email.format(x),
+					first_name=username.format(x),
+				)).insert()
+
+			# Ensure the next check should fail
+			self.assertGreater(limit, frappe.local.conf.get("throttle_user_limit", 60))
+
+			# Check that throttle_user_limit has been reached
+			with self.assertRaises(frappe.exceptions.ValidationError):
+				frappe.flags.in_import = False  # enable throttling
+				sign_up('test_registration@example.com', 'Test registration user', None)
+
+			frappe.flags.in_import = True # disable throttling
+
+			# Check that signup rate limit has been reached
+			sign_up('test_registration@example.com', 'Test registration user', None)
+			self.assertEqual(frappe.local.response['http_status_code'], 429)
```

Comment:
Reading this test made me realize we probably need to make this limit configurable, rather than having a hardcoded limit.

Can you add a config key say `throttle_signup_limit` with a fallback value as 300, similar to `throttle_user_limit` is used in `throttle_user_creation`? 

https://github.com/frappe/frappe/blob/f37789280b9f75fffc7e392da28da00c635a9ab9/frappe%2Fcore%2Fdoctype%2Fuser%2Fuser.py#L1036

Through that, we can lower this limit and will have to create only a couple of users instead of 300

Reply:
Yes, can I implement this feature in another pull-request ?